docker/Dockerfile

```diff
@@ -1,42 +1,38 @@
-# This should match the default Python version in the test actions
+# Ensure builder and runtime use same python binary
 ARG python_version=3.8
-
-FROM python:${python_version} as builder
-# netifaces doesn't have wheels built for Python >3.6; build it ourselves
-# and then install in the next stage.
-# yappi also doesn't have any wheel built.
-# Added wheel for psutil to make ARM64 happy
-
-RUN pip wheel \
-  --wheel-dir /custom-wheels \
-  netifaces \
-  yappi \
-  psutil
-
-FROM python:${python_version}-slim as base
-
+FROM python:${python_version}-slim-buster as python-base
+ENV VIRTUAL_ENV=/venv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+ENV POETRY_VERSION=1.1.5
+RUN pip install "poetry==$POETRY_VERSION"
+
+# Don't buffer output to stdout
+ENV PYTHONUNBUFFERED=1
+
+# First stage to compile deps
+FROM python-base as venv-builder
+
+RUN apt update && apt install -y --no-install-recommends \
+    gcc \
+    linux-libc-dev \
+    libc6-dev
+
+RUN python -m venv $VIRTUAL_ENV
```

Comment:
Doesn't poetry install/manage its own virtual environment by default? How does poetry "know" to install things here? I don't see $VIRTUAL_ENV being used anywhere else.

Reply:
Poetry knows about a number of env variables.

```env
ENV VIRTUAL_ENV=/venv
ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```
Is what `<venv>/bin/activate` does, minus the shell prompt. Poetry and pip both understand this, and will re-use an existing venv at the appropriate path.

Reply:
I agree on the developer side overhead for running tests inside/outside the container.
This more with the idea that since we were taking the trouble to generate a venv for the tests, and a venv for the build process, maybe it made sense to use the same one for both.